### ISSUE TITLE
Live coding and development use case – How to avoid user action for development ?

### ISSUE THREAD

**Issue by TiborUdvari** [immersive-web/navigation#17]
I'm wondering if there is a way, after the user launches a WebXR session on a page, for that page to consequently launch the immersive experience without having to ask for permission again. The info on this seems a little outdated, but [according to this](https://github.com/aframevr/aframe/issues/4445), it seems to be supported on the Quest browser only for now. I can't find any other place where it might be documented either. I would expect it to be [here](https://wpt.fyi/results/webxr?label=master&label=experimental&aligned), for example.

I'm using web editors that rely on iframes, so I'm trying to request the permission from the parent of the iframe without success for now, and then I stumbled on this navigation proposal, which seems to be a simpler way of getting this to work. 

After a quick glance, I've seen most discussions being about navigating to external or different pages. My question is specifically this:

## TLDR Question

Is it possible today to not have to request permission via a user action on a page where the user has already granted permission?

* Targeting a dev use case, it should work with the Immersive Web Emulator + polyfill

## Use Case 1 - Live Coding

I'm working on tools to get live coding working with WebXR. It works well, except when the whole program needs to be relaunched. This can happen if the user makes a mistake we can't recover from or wants to preload new resources.

[https://github.com/ffd8/P5LIVE/assets/1434442/535cf140-93d1-40bc-909b-3ac5043359ff](https://github.com/ffd8/P5LIVE/assets/1434442/535cf140-93d1-40bc-909b-3ac5043359ff)

Source: [https://github.com/ffd8/P5LIVE/issues/88#issuecomment-2020715168](https://github.com/ffd8/P5LIVE/issues/88#issuecomment-2020715168)

## Use Case 2 - Less Friction in Dev Experience

Whenever users change something in the webpage, if they aren't using modern hot reloading tech, they have to grant permission by clicking a button each time there is a change. This slows down iteration, and it's not great!

There are probably workarounds for this, but having a way to handle it out of the box would be ideal.

![Screenshot 2024-09-05 at 14:29:32](https://github.com/user-attachments/assets/1d93b54a-8331-440a-a9e0-902df701ab40)
